src/security/pausable.cairo

```diff
@@ -38,30 +45,48 @@ mod Pausable {
         const NOT_PAUSED: felt252 = 'Pausable: not paused';
     }
 
-    #[external(v0)]
-    impl PausableImpl of super::IPausable<ContractState> {
-        fn is_paused(self: @ContractState) -> bool {
+    #[embeddable_as(PausableImpl)]
+    impl Pausable<
+        TContractState, +HasComponent<TContractState>
+    > of super::IPausable<ComponentState<TContractState>> {
+        /// Returns true if the contract is paused, and false otherwise.
+        fn is_paused(self: @ComponentState<TContractState>) -> bool {
             self.Pausable_paused.read()
         }
     }
 
-    #[generate_trait]
-    impl InternalImpl of InternalTrait {
-        fn assert_not_paused(self: @ContractState) {
+    trait InternalTrait<TContractState> {
+        fn assert_not_paused(self: @TContractState);
+        fn assert_paused(self: @TContractState);
+        fn _pause(ref self: TContractState);
+        fn _unpause(ref self: TContractState);
+    }
+
+    impl InternalImpl<
+        TContractState, +HasComponent<TContractState>
+    > of InternalTrait<ComponentState<TContractState>> {
+        /// Makes a function callable when the contract is not paused.
+        fn assert_not_paused(self: @ComponentState<TContractState>) {
             assert(!self.Pausable_paused.read(), Errors::PAUSED);
         }
 
-        fn assert_paused(self: @ContractState) {
+        /// Makes a function callable when the contract is paused.
+        fn assert_paused(self: @ComponentState<TContractState>) {
             assert(self.Pausable_paused.read(), Errors::NOT_PAUSED);
         }
 
-        fn _pause(ref self: ContractState) {
+        /// Triggers a stopped state. The contract must not already be paused.
+        /// Emits a [Paused](Paused) event.
```

Comment:
Is this syntax for events in comments documented?

Reply:
No, good catch. Will update